app/grandchallenge/components/tasks.py

```diff
@@ -1191,6 +1191,25 @@ def validate_voxel_values(*, civ_pk):
     civ.interface._validate_voxel_values(civ.image)
 
 
+def _get_object_with_lock(*, app_label, model_name, object_pk):
+    from grandchallenge.archives.models import ArchiveItem
+    from grandchallenge.reader_studies.models import DisplaySet
+
+    try:
+        object = lock_model_instance(
+            app_label=app_label, model_name=model_name, pk=object_pk
+        )
+    except ObjectDoesNotExist as e:
+        if model_name in [
+            ArchiveItem._meta.model_name,
+            DisplaySet._meta.model_name,
+        ]:
+            return None
+        else:
+            raise e
```

Comment:
suggestion, but not important:
```suggestion
    except (ArchiveItem.ObjectDoesNotExist, DisplaySet.ObjectDoesNotExist):
        return None
```
Also fyi, you can use raise without having to define "e". It will reraise the exception.

Reply:
Far cleaner, nice suggestion!